contracts/BC_fusion/StakeHub.sol

```diff
@@ -0,0 +1,807 @@
+     */
+    function claim(
+        address operatorAddress,
+        uint256 requestNumber
+    ) external whenNotPaused notInBlackList validatorExist(operatorAddress) {
+        uint256 bnbAmount = IStakeCredit(_validators[operatorAddress].creditContract).claim(msg.sender, requestNumber);
+        emit Claimed(operatorAddress, msg.sender, bnbAmount);
+    }
+
+    function sync(address[] calldata operatorAddresses, address account) external whenNotPaused notInBlackList {
+        uint256 _length = operatorAddresses.length;
+        address[] memory stakeCredits = new address[](_length);
+        address _pool;
+        for (uint256 i = 0; i < _length; ++i) {
+            _pool = _validators[operatorAddresses[i]].creditContract;
+            require(_pool != address(0), "validator not exist");
+            stakeCredits[i] = _pool;
+        }
+
+        IGovToken(GOV_TOKEN_ADDR).sync(stakeCredits, account);
+    }
+
+    /*----------------- system functions -----------------*/
+    function distributeReward(address consensusAddress) external payable onlyValidatorContract {
+        address operatorAddress = _consensusToOperator[consensusAddress];
+        Validator memory valInfo = _validators[operatorAddress];
+        if (valInfo.creditContract == address(0) || valInfo.jailed) {
+            emit RewardDistributeFailed(operatorAddress, "INVALID_VALIDATOR");
+            return;
+        }
+
+        IStakeCredit(valInfo.creditContract).distributeReward{ value: msg.value }(valInfo.commission.rate);
+        emit RewardDistributed(operatorAddress, msg.value);
+    }
+
+    function downtimeSlash(address consensusAddress) external onlySlash {
+        address operatorAddress = _consensusToOperator[consensusAddress];
+        require(operatorAddress != address(0), "INVALID_CONSENSUS_ADDRESS"); // should never happen
+        Validator storage valInfo = _validators[operatorAddress];
+        require(valInfo.creditContract != address(0), "VALIDATOR_NOT_EXIST");
+
+        // slash
+        uint256 slashAmount = IStakeCredit(valInfo.creditContract).slash(downtimeSlashAmount);
+        uint256 jailUntil = block.timestamp + downtimeJailTime;
+        _jailValidator(valInfo, jailUntil);
+
+        emit ValidatorSlashed(operatorAddress, jailUntil, slashAmount, SlashType.DownTime);
+
+        _sync(valInfo.creditContract, operatorAddress);
+    }
+
+    function maliciousVoteSlash(bytes calldata _voteAddr) external onlySlash {
+        address operatorAddress = _voteToOperator[_voteAddr];
+        require(operatorAddress != address(0), "INVALID_CONSENSUS_ADDRESS"); // should never happen
+        Validator storage valInfo = _validators[operatorAddress];
+        require(valInfo.creditContract != address(0), "VALIDATOR_NOT_EXIST");
+
+        // check if can be jailed
+        // there are two slots for each day
+        uint256 dayIndex1 = (block.timestamp / 1 days) * 2;
+        uint256 dayIndex2 = dayIndex1 + 1;
```

Comment:
If we allow 10 validators to be slashed in one day, then we should have `dayIndex10`?

I dont encourage such implementation.

Please make it simple..  `    mapping(uint256 => uint256) private jailedBitMap;`, key is the day, value is the total malicous slash + double sign slash.

Reply:
Done

Reply:
`uint256 dayIndex = block.timestamp;` This is wrong